ec2/spark_ec2.py

```diff
@@ -502,8 +502,10 @@ def launch_cluster(conn, opts, cluster_name):
                                 if opts.additional_security_group in (sg.name, sg.id)]
     print "Launching instances..."
 
+    # Check if the AMI exists
     try:
-        image = conn.get_all_images(image_ids=[opts.ami])[0]
+        if conn.get_image(image_id=opts.ami) is None:
+            raise Exception()
```

Comment:
I would say not. The exception handling is hard to follow in this flow due to the bug in boto.